save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/TopBarUrl.kt

```diff
@@ -0,0 +1,158 @@
+package com.saveourtool.save.frontend.utils
+
+import com.saveourtool.save.entities.benchmarks.BenchmarkCategoryEnum
+import com.saveourtool.save.validation.FrontendRoutes
+
+/**
+ * The class for analyzing url address and creating links in topBar
+ * @property href
+ */
+class TopBarUrl(val href: String) {
+    /**
+     * CurrentPath is the link that we put in buttons
+     */
+    var currentPath = "#"
+    private var circumstance: SituationUrlClassification = SituationUrlClassification.KEYWORD_PROCESS
+    private var processLastSegments = 0
+    private val sizeUrlSegments: Int = href.split("/").size
+
+    init {
+        findExclude(href)
+    }
+
+    /**
+     * The function is called to specify the link address in the button before creating the button itself
+     *
+     * @param pathPart is an appended suffix to an already existing [currentPath]
+     */
+    fun changeUrlBeforeButton(pathPart: String) {
+        currentPath = when (circumstance) {
+            SituationUrlClassification.PROJECT, SituationUrlClassification.ORGANIZATION -> "#/${FrontendRoutes.PROJECTS.path}"
+            SituationUrlClassification.ARCHIVE -> "#/${FrontendRoutes.AWESOME_BENCHMARKS.path}"
+            SituationUrlClassification.DETAILS, SituationUrlClassification.EXECUTION -> if (pathPart == "execution") currentPath else mergeUrls(pathPart)
+            else -> mergeUrls(pathPart)
+        }
+    }
+
+    /**
+     * The function is called to specify the link address in the button after creating the button itself
+     *
+     * @param pathPart is an appended suffix to an already existing [currentPath]
+     */
+    fun changeUrlAfterButton(pathPart: String) {
+        fixCurrentPathAfter(pathPart)
+        fixExcludeAfter("\\d+".toRegex().matches(pathPart))
+    }
+
+    /**
+     * The function set a flag whether to create this button or not
+     *
+     * @param index
+     */
+    fun isCreateButton(index: Int) = when (circumstance) {
+        SituationUrlClassification.KEYWORD_PROCESS_LAST_SEGMENTS -> index >= sizeUrlSegments - 1 - processLastSegments
+        SituationUrlClassification.KEYWORD_NOT_PROCESS -> false
+        else -> true
+    }
+
+    /**
+     * The function classification url address
+     *
+     * @param href
+     */
+    private fun findExclude(href: String) {
+        circumstance = if (href.contains(SituationUrlClassification.ORGANIZATION.regex)) {
```

Comment:
Suggestion: change `regex` to nullable with null as default value and then it can be changed:
```kotlin
SituationUrlClassification.values()
.filter { it.regex != null }
.firstOrNull { href.contains(it.regex) }
?: SituationUrlClassification.KEYWORD_PROCESS
```

Reply:
Done